src/Task.test.js

```diff
@@ -1,31 +1,77 @@
+      expect(tasks).toHaveLength(2);
+      expect(tasks[0].description).toBe('Task 1');
+      expect(tasks[1].description).toBe('Task 2');
     });
   });
-  describe('DeleteTask(id)', () => {
-    it('should delete the task with the given id', () => {
-      taskso.tasks.push(
-        { id: 1, description: 'Task 1', iscomplete: false },
-        { id: 2, description: 'Task 2', iscomplete: true },
-        { id: 3, description: 'Task 3', iscomplete: false },
-      );
-      const taskIdToDelete = 2;
-      taskso.DeleteTask(taskIdToDelete);
-      expect(taskso.tasks).toHaveLength(2);
-      expect(taskso.tasks.some((task) => task.id === taskIdToDelete)).toBeFalsy();
+  describe('DeleteTask', () => {
+    it('should delete the task with the specified id', () => {
+      const taskso = new Tasks();
+      taskso.AddTask('Task 1');
+      taskso.AddTask('Task 2');
+      taskso.AddTask('Task 3');
+      let tasks = JSON.parse(localStorage.getItem('tasks'));
+      taskso.DeleteTask(2);
+      tasks = JSON.parse(localStorage.getItem('tasks'));
+      expect(tasks).toHaveLength(2);
+      expect(tasks.map((task) => task.description)).toEqual(['Task 1', 'Task 3']);
+    });
+    it('should not delete anything if the id does not exist', () => {
+      const taskso = new Tasks();
+      taskso.AddTask('Task 1');
+      taskso.AddTask('Task 2');
+      let tasks = JSON.parse(localStorage.getItem('tasks'));
+      taskso.DeleteTask(10);
+      tasks = JSON.parse(localStorage.getItem('tasks'));
+      expect(tasks).toHaveLength(2);
+      expect(tasks.map((task) => task.description)).toEqual(['Task 1', 'Task 2']);
+    });
+  });
+  describe('EditTask', () => {
+    it('should update the description of the task with the specified id', () => {
+      const taskso = new Tasks();
+      taskso.AddTask('Task 1');
+      let tasks = JSON.parse(localStorage.getItem('tasks'));
+      taskso.EditTask(1, 'Updated Task');
+      tasks = JSON.parse(localStorage.getItem('tasks'));
+      expect(tasks[0].description).toBe('Updated Task');
+    });
+    it('should not update anything if the id does not exist', () => {
+      const taskso = new Tasks();
+      taskso.AddTask('Task 1');
+      let tasks = JSON.parse(localStorage.getItem('tasks'));
+      taskso.EditTask(10, 'Updated Task');
+      tasks = JSON.parse(localStorage.getItem('tasks'));
+      expect(tasks[0].description).toBe('Task 1');
+    });
+  });
+  describe('CheckTask', () => {
+    it('should update the iscomplete status of the task with the specified id', () => {
+      const taskso = new Tasks();
+      taskso.AddTask('Task 1');
+      let tasks = JSON.parse(localStorage.getItem('tasks'));
+      taskso.CheckTask(1, true);
+      tasks = JSON.parse(localStorage.getItem('tasks'));
+      expect(tasks[0].iscomplete).toBe(true);
+    });
+    it('should not update anything if the id does not exist', () => {
+      const taskso = new Tasks();
+      taskso.AddTask('Task 1');
```

Comment:
- [ ] Please from the project requirement, You should create a test file(s) ([..].test.js) for a file(s) containing the following functions: So I suggest you create create a new test file for these functions :blush: 
    
    -   a function for editing the task description.
    -   a function for updating an item's 'completed' status.
    -   the "Clear all completed" function 

Please note that currently, you are missing a test for the `ClearCompletedTasks` function so make sure that write test for it :pray: